lib/cog/command/pipeline/executor.ex

```diff
@@ -387,26 +317,39 @@ defmodule Cog.Command.Pipeline.Executor do
   # Given pipeline output, apply templating as appropriate for each
   # adapter/destination it is to be sent to, and send it to each.
   defp respond(%__MODULE__{}=state) do
-    destinations = state.destinations
-    adapters = Map.keys(destinations)
-    bundle = state.current_plan.command.bundle
     output = state.output
-    rendered = render_for_adapters(adapters, bundle, output)
+    bundle = state.current_plan.command.bundle
+    by_output_level = Enum.group_by(state.destinations, &(&1.output_level))
 
-    case rendered do
+    # Render full output first
+    full = Map.get(by_output_level, :full, [])
+    adapters = full |> Enum.map(&(&1.adapter)) |> Enum.uniq
+
+    case render_for_adapters(adapters, bundle, output) do
       {:error, {error, template, adapter}} ->
+        # TODO: need to send error, THEN fail at the end, since we may
+        # need to do it for status-only destinations
         fail_pipeline_with_error({:template_rendering_error, {error, template, adapter}}, state)
       messages ->
-        messages
-        |> consolidate_by_adapter(destinations)
-        |> Enum.each(fn({msg, adapter, destination}) ->
-          publish_response(msg, destination, adapter, state)
+        Enum.each(full, fn(dest) ->
+          msg = Map.fetch!(messages, dest.adapter)
+          publish_response(msg, dest.room, dest.adapter, state)
         end)
     end
+
+    # Now for status only
+    # No rendering, just status map
+    by_output_level
+    |> Map.get(:status_only, [])
+    |> Enum.each(fn(dest) ->
+      publish_response(%{status: "ok"}, dest.room, dest.adapter, state)
+    end)
+
   end
 
   defp succeed_with_response(state) do
     respond(state)
+    # TODO: what happens if we fail due to a template error?
```

Comment:
This is a good question to answer. I'd imagine there will be more than a few first time devs stubbing their toes on template typos.